src/main/java/seedu/trackbeau/logic/parser/FindCommandParser.java

```diff
@@ -19,67 +25,80 @@
-        if (!matcher.matches()) {
-            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, HelpCommand.MESSAGE_USAGE));
+        if (argMultimap.getValue(PREFIX_NAME).isPresent() && argMultimap.getPreamble().isEmpty()) {
+            prefixArr.add(0,
+                    Arrays.asList(ParserUtil
+                            .parseName(argMultimap.getValue(PREFIX_NAME).get()).toString().split(" ")));
         }
 
-        splitUserInput[0] = matcher.group("searchAreaKeyWord");
-        splitUserInput[1] = matcher.group("arguments").trim();
+        if (argMultimap.getValue(PREFIX_PHONE).isPresent() && argMultimap.getPreamble().isEmpty()) {
+            prefixArr.add(1,
+                    Arrays.asList(ParserUtil
+                            .parseName(argMultimap.getValue(PREFIX_PHONE).get()).toString().split(" ")));
+        }
 
-        if (splitUserInput[1].isEmpty()) {
-            throw new ParseException(
-                    String.format(MESSAGE_INVALID_COMMAND_FORMAT, FindCommand.MESSAGE_USAGE));
+        if (argMultimap.getValue(PREFIX_EMAIL).isPresent() && argMultimap.getPreamble().isEmpty()) {
+            prefixArr.add(2,
+                    Arrays.asList(ParserUtil
+                            .parseName(argMultimap.getValue(PREFIX_EMAIL).get()).toString().split(" ")));
+        }
+
+        if (argMultimap.getValue(PREFIX_ADDRESS).isPresent() && argMultimap.getPreamble().isEmpty()) {
+            prefixArr.add(3,
+                    Arrays.asList(ParserUtil
+                            .parseName(argMultimap.getValue(PREFIX_ADDRESS).get()).toString().split(" ")));
         }
 
-        return splitUserInput;
+        if (argMultimap.getValue(PREFIX_SKINTYPE).isPresent() && argMultimap.getPreamble().isEmpty()) {
+            prefixArr.add(4,
+                    Arrays.asList(ParserUtil
+                            .parseName(argMultimap.getValue(PREFIX_SKINTYPE).get()).toString().split(" ")));
+        }
+
+        if (argMultimap.getValue(PREFIX_HAIRTYPE).isPresent() && argMultimap.getPreamble().isEmpty()) {
+            prefixArr.add(5,
+                    Arrays.asList(ParserUtil
+                            .parseName(argMultimap.getValue(PREFIX_HAIRTYPE).get()).toString().split(" ")));
+        }
+
+        if (argMultimap.getValue(PREFIX_STAFFS).isPresent() && argMultimap.getPreamble().isEmpty()) {
+            prefixArr.add(6,
+                    Arrays.asList(ParserUtil
+                            .parseName(argMultimap.getValue(PREFIX_STAFFS).get()).toString().split(" ")));
+        }
+
+        if (argMultimap.getValue(PREFIX_SERVICES).isPresent() && argMultimap.getPreamble().isEmpty()) {
+            prefixArr.add(7,
+                    Arrays.asList(ParserUtil
+                            .parseName(argMultimap.getValue(PREFIX_SERVICES).get()).toString().split(" ")));
+        }
+
+        if (argMultimap.getValue(PREFIX_ALLERGIES).isPresent() && argMultimap.getPreamble().isEmpty()) {
```

Comment:
codes for these chunks are very similar - you can consider abstracting out to another method